### ISSUE TITLE
Source mappings for CSS not showing up in dev tools

### ISSUE THREAD

**Issue by Parcye** [understrap/understrap#268]
I see the .min.css.map file updated on every change, but if I look in dev tools on my theme, The CSS file also has the correct file named below : 
/*# sourceMappingURL=theme.min.css.map */

But still... it isn't showing up. How can this happen?

**Comment by rzariy** [comment]
add .pipe(sourcemaps.init()) and .pipe(sourcemaps.write('./')) in gulp task sass

`gulp.task('sass', function () {
    var stream = gulp.src('./sass/*.scss')
        .pipe(plumber())
	.pipe(sourcemaps.init()) // add this
        .pipe(sass())
	.pipe(sourcemaps.write('./')) // add this
        .pipe(gulp.dest('./css'))
        .pipe(rename('custom-editor-style.css'))
        .pipe(gulp.dest('./css'));
    return stream;
});`

**Comment by SergiArias** [comment]
I have the same problem, but the solution provided doesn't work for me. rzariy, are you running gulp watch-bs or gulp watch-scss?

**Comment by rzariy** [comment]
@SergiArias I'm running gulp watch-bs but when i'm running watch-scss, it's works also.
I'm not touching any line in gulpfile.js exept that, but sourcemap not work correctly when, W3 total cache plugins active, so i' deactivated it when development stage.

**Comment by SergiArias** [comment]
@rzariy I think it's related to the minify process, which happens after sass process. Did you try it with last version?

**Comment by SergiArias** [comment]
I tried the solution in the 0.6 release instead of master, and it doesn't work for me either, but thank you @rzariy .
What solved the problem temporarily is to change to the non minified version through header.php, which works with sourcemaps.

**Comment by SergiArias** [comment]
I just found a more elegant solution until it gets solved. Just append this code at the end of the gulp file. I took out the minimizing so it loads faster, as it's only for development.

`gulp.task('cssnano-sergi', function(){
  return gulp.src('./css/theme.css')
    .pipe(sourcemaps.init({loadMaps: true}))
    .pipe(plumber())
    .pipe(rename({suffix: '.min'}))
    .pipe(sourcemaps.write('./'))
    .pipe(gulp.dest('./css/'))
});
gulp.task('watch-nano', function () {
    gulp.watch('./sass/**/*.scss', ['scss-for-dev']);
    gulp.watch('./css/theme.css', ['cssnano-sergi']);
    gulp.watch([basePaths.dev + 'js/**/*.js','js/**/*.js','!js/theme.js','!js/theme.min.js'], ['scripts']);

    //Inside the watch task.
    gulp.watch('./img/**', ['imagemin'])
});
gulp.task('watch-sergi', ['browser-sync', 'watch-nano', 'cssnano-sergi', 'scripts'], function () { });
gulp.task('default', ['watch-sergi']);`

**Comment by rzariy** [comment]
hi @SergiArias I think I found the solution, sourcemap should created before changing suffix to .min.
in older versioan it's using cssnano but in newer it's using minify-css, and there is missing task to creating sourcemap.
since I'm new in gulp I don't know it's best practice workflow or not, but it's work.


here my code:
<pre>
// Run:
// gulp sass + cssnano + rename
// Prepare the min.css for production (with 2 pipes to be sure that "theme.css" == "theme.min.css")
gulp.task('scss-for-prod', function() {
    var source =  gulp.src('./sass/*.scss')
        .pipe(plumber())
        .pipe(sourcemaps.init({loadMaps: true}))
        .pipe(sass());
	

    var pipe1 = source.pipe(clone())
        .pipe(sourcemaps.write(undefined, { sourceRoot: null }))
        .pipe(gulp.dest('./css'))
        .pipe(rename('custom-editor-style.css'))
	.pipe(gulp.dest('./css')); // add this


    var pipe2 = source.pipe(clone())
	.pipe(plumber()) // add this
        .pipe(minify-css())
        .pipe(rename({suffix: '.min'}))
        .pipe(gulp.dest('./css'));

    return merge(pipe1, pipe2);
});
</pre>

<pre>
gulp.task('sass', ['minify-css'], function () {
    var stream = gulp.src('./sass/*.scss')
        .pipe(plumber())
	.pipe(sourcemaps.init()) // add this
        .pipe(sass())
	.pipe(sourcemaps.write('./', {includeContent: false})) // add this
	.pipe(sourcemaps.init({loadMaps: true})) // add this
        .pipe(gulp.dest('./css'))
        .pipe(rename('custom-editor-style.css'))
	.pipe(gulp.dest('./css')); // add this
    return stream;
});
</pre>

<pre>
  return gulp.src('./css/theme.css')
        .pipe(cleanCSS({compatibility: 'ie8'}))
	.pipe(sourcemaps.init({loadMaps: true})) // add this
	.pipe(plumber())
	.pipe(rename({suffix: '.min'}))
	.pipe(sourcemaps.write('./')) // add this
        .pipe(gulp.dest('./css/'));
});
</pre>

**Comment by SergiArias** [comment]
Yes! I think this is the solution they should apply to next version!

**Comment by seangreen** [comment]
Not working for me. :(

**Comment by SergiArias** [comment]
seangreen are you in last version? Did you add all the `// add this` lines?

**Comment by nicholasdixon** [comment]
When will this issue be fixed?
I only downloaded Understrap about 10 days ago and my code looks a bit different to the examples rzariy posted, but the sourcemaps are still not working, so I am unsure how to fix this issue?

**Comment by Glenn-Domin** [comment]
Yeah there have been some changes to that file since the ticket has been opened, though the issue persists. I've done a fair few sites using understrap and there are two lines that need changing. I don't currently have access to those projects but I have the following in my old notes - hopefully they still work.
Replace this line `.pipe(sass())` with:
```
.pipe(sourcemaps.init({loadMaps: true}))
.pipe(sass())
.pipe(sourcemaps.write(undefined, { sourceRoot: null }))
```

**Comment by axlright** [comment]
I fiddled around and found what I think is a fix. If people can confirm that it works for them, I can post it as a pull request.

In the gulpfile.js, change:
```
// Run:
// gulp sass
// Compiles SCSS files in CSS
gulp.task('sass', function () {
    var stream = gulp.src('./sass/*.scss')
        .pipe(plumber({
            errorHandler: function (err) {
                console.log(err);
                this.emit('end');
            }
        }))
        .pipe(sass())
        .pipe(gulp.dest('./css'))
        .pipe(rename('custom-editor-style.css'))
    return stream;
});
```
to
```
// Run:
// gulp sass
// Compiles SCSS files in CSS
gulp.task('sass', function () {
    var stream = gulp.src('./sass/*.scss')
        .pipe(sourcemaps.init({loadMaps: true}))
        .pipe(plumber({
            errorHandler: function (err) {
                console.log(err);
                this.emit('end');
            }
        }))
        .pipe(sass())
        .pipe(sourcemaps.write('./'))
        .pipe(gulp.dest('./css'))
        .pipe(rename('custom-editor-style.css'))
    return stream;
});
```
Lemme know any comments. Been meaning to look for a solution to this for a while.

**Comment by axlright** [comment]
The above solution creates both a css.map and a css.min.map which isn't perfect but it does work in Chrome dev tools...

**Comment by Glenn-Domin** [comment]
Have managed to dig up one of my working gulpfiles. Seems similar to the above. Here's the relevant part:
```
// Run:
// gulp sass
// Compiles SCSS files in CSS
gulp.task('sass', function () {
    var stream = gulp.src('./sass/*.scss')
        .pipe(plumber({
            errorHandler: function (err) {
                console.log(err);
                this.emit('end');
            }
        }))
        .pipe(sourcemaps.init({loadMaps: true}))
        .pipe(sass())
        .pipe(sourcemaps.write(undefined, { sourceRoot: null }))
        .pipe(gulp.dest('./css'))
        .pipe(rename('custom-editor-style.css'))
    return stream;
});
```

**Comment by axlright** [comment]
This works better - doesn't create an unnecessary css.map and works to map all the css.min styles back to the original _file.scss files. Nice!

Someone should put this through as a pull request. I'm happy to but I'm not the one who made the final fix Having the source mapping fixed is a huge help.

**Comment by SergiArias** [comment]
I recently switched to [roots sage](https://roots.io/sage/) and I cannot recommend it enough. All this little bugs don't exist, it's much more ordered and faster as it doesn't use gulp anymore. Give it a try!!

**Comment by axlright** [comment]
Solution for this issue has been tested with `gulp watch` and `gulp watch-bs`

I was having trouble with `gulp watch-bs` on a local url running https:// - [source maps using local ssl certificates seems to have been an issue with BrowserSync](https://github.com/BrowserSync/browser-sync/issues/639).

I found a solution and regardless it's not an issue with UnderStrap's application. I'll make sure the Pull Request is pushed through.

Thanks @Glenn-Domin and @ezrx and everyone who brought up the issue

**Comment by axlright** [comment]
I've just added a pull request #607 with a solution that you all helped push forward. Anyone is welcome to test and we'll get it into the next release.

Thanks again @Glenn-Domin and @ezrx

**Comment by holger1411** [comment]
Looks good here. Merged it and will be shipped with 0.8.2
Let me know if there is feedback/bug reports and feel free to re-open this
